influxql/iterator.go

```diff
@@ -618,6 +618,8 @@ func (a IteratorCreators) CreateIterator(opt IteratorOptions) (Iterator, error)
 			itr, err := ic.CreateIterator(opt)
 			if err != nil {
 				return err
+			} else if itr == nil {
```

Comment:
@jsternberg I added a `nil` return if there's no iterators.

@e-dard I agree that a list of invariants would be good. That's a larger project in and of itself.